### PR TITLE
DatePicker UX Implementations

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#8913

*Description of changes:*
* Imported/applied update DatePicker component
* Minor prop updates for new DatePicker Version

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by argush3 — APPROVED** [review]

File: src/rules/date-rules.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@
 // and matching what date rules to use.
 export const DateRules = {
   REQUIRED: (v: string) => !!v || 'Select date',
+  RESOLUTION_REQUIRED: (v: string) => !!v || 'Resolution date is required',
+  SIGNATURE_REQUIRED: (v: string) => !!v || 'Signature date is required',
```
Comment (severinbeauvais) [inline src/rules/date-rules.ts:7]: I'm thinking these don't belong in DateRules...

Maybe under ResolutionRules and SignatureRules ?

**Review by severinbeauvais — APPROVED** [review]
Some ideas but no show stoppers.

File: src/rules/date-rules.ts (hunk 1 above)
Comment (cameron-freshworks) [reply src/rules/date-rules.ts:7]: You are right, too specific! Will fix this

Comment (cameron-freshworks) [reply src/rules/date-rules.ts]: Updated. Maybe future state we figure out how to pass arguments into the rules to make the return messages custom.

**Pull request merged**